clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/containers/KubernetesCluster.java

```diff
@@ -197,4 +209,49 @@ private String runKindCmd(String args) throws IOException, InterruptedException
         .isEqualTo(0);
     return output;
   }
+
+  /**
+   * Returns the kubectl version to use for the given kubernetes version. This is used to download
+   * the correct kubectl binary for the cluster.
+   *
+   * @return The kubectl version to use.
+   */
+  private static String getKubectlVersion() {
```

Comment:
I'm torn about this.  Part of me really likes it.  Part of me thinks we end up testing something farther away than what we ship, because the default version of kubectl is the only one that's used by default.

Does it make sense to change the logic e.g. [here](https://github.com/spinnaker/clouddriver/blob/069140bae4cc171f9d0e85b7ec40141affd4b5b9/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutor.java#L698-L702) to dynamically choose the kubectl binary that matches the target cluster version?  Which I suppose would then mean on startup / periodically we'd have to query target clusters for their version?

Reply:
I'd love that feature! It would mean that the scope of this PR grows substantially, however... Do you want me to give it a go? What k8s versions would we support then? Also I haven't found a good way to automatically download the latest patch version of a given `kubectl` release. It would be awesome to not have to manually maintain that kubectl list in all of the dockerfiles and the deb packaging code.

Reply:
I wasn't thinking about dynamically downloading kubectl binaries based on clusters we discover.  I suppose that's a possibility, but could be tough to keep compliance folks happy.

I was more thinking about choosing from among the ones we've got.  It probably does belong in another PR.

For this PR, what do you think of continuing to test the default kubectl version against all the k8s cluster versions, and then one more round of testing using the mapping here, and if we're very lucky, removing dups?  That's way less than every client x every server, but also doesn't reduce the coverage we're getting today.

Then if we ever do dynamically choose the k8s version, we could drop the default round of testing.

One other thought that's been rolling around.  I'd love to have another step in the PR build github action that invokes a new gradle Test task to exercise the just-built docker image with testcontainers.  Not sure what the magic is to learn the name and tag, or perhaps we get to specify the tag and then know what to tell gradle.

With this, we could potentially adjust these tests to use the kubectl binaries in the docker image and not have to download them.

Reply:
> For this PR, what do you think of continuing to test the default kubectl version against all the k8s cluster versions, and then one more round of testing using the mapping here, and if we're very lucky, removing dups? That's way less than every client x every server, but also doesn't reduce the coverage we're getting today. 

I've removed the in-code kubectl selection stuff and reverted to just using GitHub Actions. It's a bit cleaner (albeit with a bit of duplication in the action matrix), and it achieves what you describe above: one round of testing with the default kubectl version and another round using the version that corresponds to the k8s version.